lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java

```diff
@@ -114,6 +114,7 @@ public static RetryStrategy getRetryStrategy(RetryMode retryMode)
     private ObjectCannedAcl objectCannedAcl = ObjectCannedAcl.NONE;
     private RetryMode retryMode = RetryMode.LEGACY;
     private int maxErrorRetries = 10;
+    private boolean supportsExclusiveCreate;
```

Comment:
If I'm reading correctly, AWS S3 and MinIO already have `If-None-Match` feature.
Why not enable it by default?

Reply:
I don't want to break existing deployments that are not using either AWS S3 or Minio.

Reply:
Let's switch it on by default after this is released

Reply:
Depending on the outcome of the conversation pls consider adapting the delta lake tests to make use of this feature if the configuration option stays `false` by default.

Reply:
I've enabled it for Minio-backed tests and added a dedicated test that runs with the concurrent reconciliation based on the exclusive create support with Minio.